src/libAtomVM/list.h

```diff
@@ -51,6 +51,8 @@ static inline void list_remove(struct ListHead *remove_item)
 {
     remove_item->prev->next = remove_item->next;
     remove_item->next->prev = remove_item->prev;
+    remove_item->next = remove_item;
+    remove_item->prev = remove_item;
```

Comment:
What is the purpose of this change? Most of the time, as soon as we remove an item, we free it just after, so doing this operation most of the time is useless.
Did we introduce any piece of code relying on this?

Reply:
It is a required change as there is a case where an entry is removed and still referred to. Entries are often removed to be added later. I am not sure if this was introduced or just revealed by the SMP code.

Reply:
Sounds like there was a mistake around, that maybe should be used using `MUTABLE_LIST_FOR_EACH`.